build/tools/ExtractRuleDescFromJson/RuleDescExtractor.cs

```diff
@@ -130,6 +130,35 @@ private void ProcessRule(PluginRule pluginRule)
         }
     }
 
+    private static Dictionary<SoftwareQuality, SoftwareQualitySeverity> ConvertDefaultImpacts(Dictionary<string, string>? defaultImpacts)
+    {
+        if (defaultImpacts != null)
+        {
+            return defaultImpacts.ToDictionary(d => ConvertSoftwareQuality(d.Key), d => ConvertSoftwareQualitySeverity(d.Value));
+        }
+
+        return null;
+    }
+
+    private static SoftwareQuality ConvertSoftwareQuality(string softwareQuality)
+    {
+        return Enum.Parse<SoftwareQuality>(softwareQuality, true);
```

Comment:
Have you looked at using this instead? https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Converters_StringEnumConverter.htm

Reply:
we are using system.text.json in this project but there's an equivalent in that also.